install.sh

```
@@ -448,14 +452,118 @@ download_and_verify() {
+  actual="$("${GENIE_HOME}/bin/genie" --version 2>/dev/null)" || return 1
+  expected_token="$(parse_version_token "$expected_version")" || return 1
+  actual_token="$(parse_version_token "$actual")" || return 1
+  [[ "$actual_token" == "$expected_token" ]]
+}
+
+# Restore the newest backup from bin/.previous over the live binary. Used only
+# when a promoted binary fails its post-swap verification.
+rollback_binary() {
+  local previous_dir="${GENIE_HOME}/bin/.previous" newest
+  [[ -d "$previous_dir" ]] || return 1
+  newest="$(ls -1t "$previous_dir"/genie-* 2>/dev/null | head -n1)"
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)